src/runtime/virtcontainers/qemu_amd64_test.go

```diff
@@ -272,6 +272,26 @@ func TestQemuAmd64AppendProtectionDevice(t *testing.T) {
 	assert.Error(err)
 	assert.Empty(bios)
 
+	// snp protection
+	amd64.(*qemuAmd64).protection = snpProtection
+
+	devices, bios, err = amd64.appendProtectionDevice(devices, firmware, "")
+	assert.NoError(err)
+	assert.Empty(bios)
+
+	expectedOut = append(expectedOut,
```

Comment:
So, this was based on my advice. In https://github.com/kata-containers/kata-containers/commit/a87698fe56631ea1f3dddf061198979e570cdbf8#diff-a267a8bb05cd3190291f375b40fa78e822b06986f6d9e88a1d68c1a0774c9447 that section was deliberately removed in `CCv0`.

Do you think we should re-add this, or can we just add `:` to the current `expectedOut = ` for snp protection to fix the compilation problem?

Reply:
The problem we have here, @stevenhorsman, is that `expectedOut` is never declared before.  So, adding a `expectedOut := append(expectedOut, ...)` will also lead to a failure.

So, what you'd need to do is:

```
expectedOut := []govmmQemu.Device{
    govmm.Qemu.Object {
        ....
    }
}
```

And then, down in the TDX part, remove the `:` from the `expectedOut` declaration.

Makes sense?

Reply:
Yep, it would be helpful if I actually looked at the code properly! The main thing is to ensure that we don't need to re-add in the SEV update, which sounds resolved.

Reply:
I think I need to append the TDX object in order to match what the test is doing. Whether that's the best approach, I'm not sure, but it's the closest to `main`. Hopefully 4th time is the charm!